contrib_bots/lib/DefineBot/DefineBot.py

```diff
@@ -0,0 +1,72 @@
+# See readme.md for instructions on running this code.
+import json
+import requests
+
+class DefineHandler(object):
+    '''
+    This plugin define a word that the inputs. It
+    looks for messages starting with '@define'.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to define a word. Users should preface
+            messages with "@define".
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        is_define = original_content.startswith('@define')
+
+        return is_define
+
+    def _handle_definition(self, original_content):
+        error_message = 'Definition not available.'
+
+        # Remove '@define' from the message and extract the rest of the message, the
+        # word/phrase to define.
+        split_content = original_content.split(' ')
+        to_define = ' '.join(split_content[1:])
+
+        # No word was entered.
+        if not to_define:
+            return 'Please enter a word/phrase to define.'
+        else:
+            response = '**%s**:\n' % (to_define)
+
+            # Some error occured with requests.get()
+            try:
+                # Use OwlBot API to fetch definition.
+                api_result = requests.get('https://owlbot.info/api/v1/dictionary/%s?format=json' % (to_define))
+                # Convert API result from string to JSON format.
+                definitions = json.loads(api_result.text)
+
+                # Could not fetch definitions for the given word/phrase.
+                if not definitions:
+                    response += error_message
+                else: # Definitions available.
+                    # Show definitions line by line.
+                    response += '\n'.join('({}) {}'.format(d['type'], d['defenition']) for d in definitions)
```

Comment:
Have you tested it visually? Sometimes a definition is not enough and usage is really helpful in understanding.